lib/core/errors.js

```diff
@@ -57,12 +57,13 @@ class BodyTimeoutError extends UndiciError {
 }
 
 class ResponseStatusCodeError extends UndiciError {
-  constructor (message, statusCode, headers) {
+  constructor (message, statusCode, headers, body) {
     super(message)
     Error.captureStackTrace(this, ResponseStatusCodeError)
     this.name = 'ResponseStatusCodeError'
     this.message = message || 'Response Status Code Error'
     this.code = 'UND_ERR_RESPONSE_STATUS_CODE'
+    this.body = body
```

Comment:
@ronag Would `responseBody` be a better name here?

Reply:
I'm ok with body